### PR TITLE
chore: add sql middleware to for warehouse

### PR DESCRIPTION
**Pull request by BonapartePC**
# Description

Add middleware to warehouse. This helps us log queries which take more time

## Notion Ticket

https://www.notion.so/rudderstacks/Apply-sql-middleware-for-warehouse-integrations-47eecf450cec45b0ba467c2b038a55e2

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/integrations/redshift/redshift.go [hunk 1]
```diff
@@ -1107,8 +1110,8 @@ func (rs *Redshift) dropDanglingStagingTables() bool {
 	return delSuccess
 }
 
-func (rs *Redshift) connectToWarehouse() (*sql.DB, error) {
-	return Connect(rs.getConnectionCredentials())
+func (rs *Redshift) connectToWarehouse() (*sqlmiddleware.DB, error) {
+	return rs.connect()
 }
```
Comment (achettyiitr) [inline warehouse/integrations/redshift/redshift.go:1115]: Let's get rid of this as just call connect?

File: warehouse/integrations/redshift/redshift.go [hunk 2]
```diff
@@ -1062,7 +1064,8 @@ func Connect(cred RedshiftCredentials) (*sql.DB, error) {
 	if err != nil {
 		return nil, fmt.Errorf("redshift set query_group error : %v", err)
 	}
-	return db, nil
+	middleware := sqlmiddleware.New(db)
+	return middleware, nil
 }
```
Comment (achettyiitr) [inline warehouse/integrations/redshift/redshift.go:1069]:
```GO
middleware := sqlmiddleware.New(
		db,
		sqlmiddleware.WithLogger(rs.Logger),
		sqlmiddleware.WithKeyAndValues(
			logfield.SourceID, rs.Warehouse.Source.ID,
			logfield.SourceType, rs.Warehouse.Source.SourceDefinition.Name,
			logfield.DestinationID, rs.Warehouse.Destination.ID,
			logfield.DestinationType, rs.Warehouse.Destination.DestinationDefinition.Name,
			logfield.WorkspaceID, rs.Warehouse.WorkspaceID,
			logfield.Schema, rs.Namespace,
		),
		sqlmiddleware.WithSlowQueryThreshold(rs.SlowQueryThreshold),
		sqlmiddleware.WithSecretsRegex(map[string]string{
			"ACCESS_KEY_ID '[^']*'":     "ACCESS_KEY_ID '***'",
			"SECRET_ACCESS_KEY '[^']*'": "SECRET_ACCESS_KEY '***'",
			"SESSION_TOKEN '[^']*'":     "SESSION_TOKEN '***'",
		}),
	)
```

File: warehouse/integrations/redshift/redshift_test.go [hunk 3]
```diff
@@ -93,9 +96,21 @@ func TestIntegrationRedshift(t *testing.T) {
...
+				Scheme: "postgres",
+				User:   url.UserPassword(credentials.Username, credentials.Password),
+				Host:   fmt.Sprintf("%s:%s", credentials.Host, credentials.Port),
+				Path:   credentials.DbName,
+			}
+			params := url.Values{}
+			params.Add("sslmode", "require")
+
+			dsn.RawQuery = params.Encode()
+			db, err := sql.Open("postgres", dsn.String())
```
Comment (achettyiitr) [inline warehouse/integrations/redshift/redshift_test.go:109]: dsn := fmt.Sprintf(
		"postgres://%s:%s@%s:%s/%s?sslmode=disable",
		credentials.User, credentials.Password, credentials.Host, credentials.Port, credentials.DBName,
	)
Maybe we can use a simpler version here?

**Review by achettyiitr — APPROVED** [review]

**Review by lvrach — APPROVED** [review]
approved, with minor comment

**Pull request merged**